README.md

```diff
@@ -93,21 +100,120 @@ Collecting logs
+running in a virtual machine via `docker-machine` and `boot2docker`. Essentially delegating to what the `docker` client
+will connect to.
+
+If you're on Mac OS X, the `docker` cli expects the following environment variables:
+
+ - `DOCKER_HOST`
+ - If the Docker daemon is secured by TLS, `DOCKER_TLS_VERIFY` and `DOCKER_CERT_PATH` need to be set.
+
+Similarly, if you're using a Local Machine, you need to ensure the Run Configuration (in your IDE, command line etc.)
+has those same variables set.
+
+An example of creating a `DockerMachine` that connects to a local docker daemon:
+
+```java
+DockerMachine.localMachine()
+             .build()
+```
+
+### Remote Machine
+
+You may not always want to connect to a Docker daemon that is running on your local computer or a virtual machine
+running on your local computer.
+
+An example of this would be running containers in a clustered manner with Docker Swarm. Since Docker Swarm implements
+the Docker API, setting the right environment variables would allow us to use Docker containers on the swarm.
+
+An example of connecting to a remote Docker daemon that has also been secured by TLS:
+
+```java
+DockerMachine.remoteMachine()
+             .host("tcp://docker-host:2376")
+             .withTLS("/path/to/cert")
+             .build()
+```
+
+### Additional Environment Variables
+
+It may also be useful to pass environment variables to the process that will call `docker-compose`.
+
+You can do so in the following manner:
+
+```java
+DockerMachine.localMachine()
+             .withEnvironmentVariable("SOME_VARIABLE", "SOME_VALUE")
+             .build()
+```
+
+The variable "SOME_VARIABLE" will be available when `docker-compose` is called, and can be used in Variable Interpolation in the compose file.
+
+### How to use a `DockerMachine`
+
+When creating a `DockerComposition` an additional parameter may be specified. That being the custom `DockerMachine`,
+by default - if no `dockerMachine` parameter is specified - `DockerComposition` will use connect to the local Docker
+daemon, similarly to how the `docker` cli works.
+
+```java
+private final DockerMachine dockerMachine = DockerMachine.localMachine()
+                                                         .withAdditionalEnvironmentVariable("SOME_VARIABLE", "SOME_VALUE")
+                                                         .build();
+
+@Rule
+DockerComposition composition = DockerComposition.of("docker-compose.yaml", dockerMachine)
+                                                 .build();
+```
+
+Composing docker compose files
+------------------------------
+
+`docker-compose` (at least as of version 1.5.0) allows us to "compose" Docker compose files. On the command line, you
+can do this with this example command:
+
+    docker-compose -f file1.yml -f file2.yml -f file3.yml
+
+Subsequent files override previous declarations of services and also add new services to the composition.
```

Comment:
Try and avoid explaining too much about how it works here, link to the docker-compose docs for more information.